script_vts.py

```diff
@@ -148,20 +87,41 @@ def data_processing(string_parse_json, res_dic, analysis_stat, res_parse_json, r
     return None, report_data_dic
 
 
-def parse_filename(filename_parsed):
-    """Function used to get a filename as input and parse it until we got a proper one"""
-    while filename_parsed.isalnum() is False:
-        try:
-            filename = input(
-                "Please provide a filename for your report (without an extension / _ is accepted between chars): ")
-            filename_parsed = filename
-            if filename != '':
-                if filename[0] != '_' and filename[-1] != '_' and ('__' in filename) is not True:
-                    filename_parsed = filename.replace('_', '')
-        except ValueError as val_err:
-            exception_handler(True, val_err)
-            return None
-    return filename
+def file_reputation_api_request(key, hash_file_reputation):
+    """Function to make api request to get a file report"""
+    url = f"https://www.virustotal.com/api/v3/files/{hash_file_reputation}"
+    headers = {"accept": "application/json",
+               "x-apikey": key
+               }
+    response_file_reputation = requests.get(url, headers=headers, timeout=10)
+    if response_file_reputation.status_code == 200:
+        return response_file_reputation.text
+    if response_file_reputation.status_code == 400:
+        print(f"[-] HTTP response wasn't successful: {response_file_reputation.status_code}, we made too many request, we need to wait 60 seconds...")
+        sleep(60)
+        file_reputation_api_request(key, hash_file_reputation)
```

Comment:
You've got an infinite loop here. If you keep getting error_code 400 will the program ever terminate?

Reply:
Yes but it is intended. Like you can see on the print statement we only receive a 400 when the API is like oh calm down you made more than 4 requests (we talked about knocking too hard on the door and that it wasnt a problem well I checked and for sure VT API is sending us a 400 error code when we make too many request where as the 401 will be used to say that our API key is not good (I'll add this feature right now). But basically we can't quit because it's stopping the whole program where as we do know that if we receive a 400 the only reason is because of this API request problem that might arise (for now there's no problem with sending multiple request but I'm trying to prevent the program to break in the future). In this case the logic I've wrote is okay I'd say but if there's a problem for you I'll find another solution right away!